app-with-redux-toolkit/src/redux/account.selectors.ts

```diff
@@ -0,0 +1,6 @@
+import {State} from "./store"
+
+export const getState = (state: State) => state.account.account;
+export const getAmount = (state: State) => state.account.account.amount;
+export const getLoading = (state: State) => state.account.account.loading;
+export const getError = (state: State) => state.account.account.error;
```

Comment:
hele, mám takovej famózní nápad...


co udělat:


```

const createSelectors = <K extends string, Selectors extends Record<K, (state: State) => any>>(selectors: Selectors) => selectors


const accountSelectors = createSelectors({
  getState: (state) => state.account.account,
  getAmount: (state) => state.account.account.amount,
  getLoading: (state) => state.account.account.loading,
  getError: (state) => state.account.account.error,
})
```

díky tomu budou všechy argumenty state automaticky otypovaný a nemusí to furt člověk typovat jak kunda



![image](https://user-images.githubusercontent.com/16742389/169041067-1f040a1a-ed4b-45d0-a7d3-68c45b55f8f7.png)


```